docs/source/class_basics.rst

```diff
@@ -194,50 +194,96 @@ effect at runtime:
-by a subclass. You can define abstract base classes using the
-``abc.ABCMeta`` metaclass, and the ``abc.abstractmethod`` and
-``abc.abstractproperty`` function decorators. Example:
+by any *concrete* (non-abstract) subclass. You can define abstract base
+classes using the ``abc.ABCMeta`` metaclass, and the ``abc.abstractmethod``
+and ``abc.abstractproperty`` function decorators. Example:
 
 .. code-block:: python
 
    from abc import ABCMeta, abstractmethod
 
-   class A(metaclass=ABCMeta):
+   class Animal(metaclass=ABCMeta):
        @abstractmethod
-       def foo(self, x: int) -> None: pass
+       def eat(self, food: str) -> None: pass
 
        @abstractmethod
-       def bar(self) -> str: pass
+       def can_walk(self) -> bool: pass
+
+   class Cat(Animal):
+       def eat(self, food: str) -> None:
+           ...  # Implementation omitted
 
-   class B(A):
-       def foo(self, x: int) -> None: ...
-       def bar(self) -> str:
-           return 'x'
+       def can_walk(self) -> bool:
+           return True
 
-   a = A()  # Error: 'A' is abstract
-   b = B()  # OK
+   x = Animal()  # Error: 'Animal' is abstract
+   y = Cat()     # OK
 
 Note that mypy performs checking for unimplemented abstract methods
 even if you omit the ``ABCMeta`` metaclass. This can be useful if the
 metaclass would cause runtime metaclass conflicts.
 
+Since you can't create instances of ABCs, they are most commonly used in
+type annotations. For example, this method accepts arbitrary iterables
+containing arbitrary animals (instances of concrete ``Animal``
+subclasses):
+
+.. code-block:: python
+
+   def feed_all(animals: Iterable[Animal], food: str) -> None:
+       for animal in animals:
+           animal.eat(food)
+
+There is one important peculiarity about how ABCs work in Python --
+whether a particular class is abstract or not is somewhat implicit.
+In the example below, ``Derived`` is treated as an abstract base class
+since ``Derived`` inherits an abstract ``f`` method from ``Base`` and
+doesn't explicitly implement it. The definition of ``Derived``
+generates no errors from mypy, since it's a valid ABC:
+
+.. code-block:: python
+
+   from abc import ABCMeta, abstractmethod
+
+   class Base(metaclass=ABCMeta):
+       @abstractmethod
+       def f(self, x: int) -> None: pass
+
+   class Derived(Base):  # No error -- Derived is implicitly abstract
+       def g(self) -> None:
+           ...
+
+Attempting to create an instance of ``Derived`` will be rejected,
+however:
+
+.. code-block:: python
+
+   d = Derived()  # Error: 'Derived' is abstract
+
+.. note::
+
+   It's a common error to forget to implement an abstract method.
+   As shown above, the class definition will not generate an error
+   in this case, but any attempt to construct an instance will be
+   flagged as an error.
+
 A class can inherit any number of classes, both abstract and
 concrete. As with normal overrides, a dynamically typed method can
```

Comment:
At some point (not now) I would like to overhaul the docs and somehow change the terminology regarding statically typed and dynamically typed functions and methods. There are flags (e.g. `--check-untyped-defs`) that blur the difference, and when we talk amongst ourselves or with users we never describe the distinction -- instead we'll use "unannotated" (even though my spell checker doesn't think that's a word :-). It's also not clear to me whether an unannotated method can still override an annotated method if the signatures differ wildly (e.g. no correspondence in argument count).